cmd/yorkie/version.go

```diff
@@ -17,27 +17,162 @@
 package main
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
 	"runtime"
 
+	"connectrpc.com/connect"
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v3"
 
+	"github.com/yorkie-team/yorkie/admin"
+	"github.com/yorkie-team/yorkie/api/types"
+	"github.com/yorkie-team/yorkie/cmd/yorkie/config"
 	"github.com/yorkie-team/yorkie/internal/version"
 )
 
+var (
+	clientOnly bool
+	output     string
+)
+
 func newVersionCmd() *cobra.Command {
 	return &cobra.Command{
-		Use:   "version",
-		Short: "Print the version number of Yorkie",
+		Use:     "version",
+		Short:   "Print the version number of Yorkie",
+		PreRunE: config.Preload,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			fmt.Printf("Yorkie: %s\n", version.Version)
-			fmt.Printf("Go: %s\n", runtime.Version())
-			fmt.Printf("Build date: %s\n", version.BuildDate)
+			if err := Validate(); err != nil {
+				return err
+			}
+
+			var versionInfo types.VersionInfo
+			versionInfo.ClientVersion = getYorkieClientVersion()
+
+			serverVersionChan := make(chan *types.VersionDetail)
+			errorChan := make(chan error)
+
+			if !clientOnly {
+				go func() {
+					rpcAddr := viper.GetString("rpcAddr")
+					auth, err := config.LoadAuth(rpcAddr)
+					if err != nil {
+						errorChan <- err
+						return
+					}
```

Comment:
@krapie I kept the go routine code. Even if an error occurs while retrieving the server version, I think it is good that it does not affect the existing behavior of showing the client version.

Reply:
The current implementation using goroutines doesn't seem to provide significant benefits:

1. Lack of asynchronous advantage: The code waits for the server response before outputting version info, so there's no real benefit from asynchronous processing.

2. Timeout handling: We can handle timeouts using context, eliminating the need for a separate goroutine.

3. Unnecessary complexity: The use of goroutines and channels adds complexity without clear benefits.

4. Error handling: The current structure complicates error handling, which could be simpler with a synchronous approach.

I suggest refactoring to a synchronous implementation, which would make the code simpler and more straightforward.